scripts/override_images/override_os_images.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Overrides a given list of OS images to a list with a single OS image
+that matches the latest release image in RELEASE_IMAGES
+"""
+import json
+import os
+import sys
+from argparse import ArgumentParser
+
+from assisted_service_client.models import OsImage, ReleaseImage
+
+from assisted_test_infra.test_infra import utils
+from consts import DEFAULT_CPU_ARCHITECTURE
+
+
+def get_os_image(
+    os_images: list[OsImage], openshift_version: str, cpu_architecture: str = DEFAULT_CPU_ARCHITECTURE
+) -> OsImage | None:
+    archs_images = [v for v in os_images if v.cpu_architecture == cpu_architecture]
+    os_images = [v for v in archs_images if v.openshift_version == openshift_version]
+    if len(os_images) >= 1:
+        return os_images[0]
+
+    return None
+
+
+def extract_version(release_image: str) -> str:
+    if not release_image:
+        return ""
+
+    full_ocp_version = utils.extract_version(release_image)
+    return f"{full_ocp_version.major}.{full_ocp_version.minor}"
+
+
+def main():
+    # Load default os images
+    with open(args.src, "r") as f:
+        os_images_parsed: list[OsImage] = json.load(f)
+
+    os_images = [OsImage(**os_image_dict) for os_image_dict in os_images_parsed]
+
+    if (release_images := os.getenv("RELEASE_IMAGES", "")) == "":
+        raise ValueError("RELEASE_IMAGES environment variable must be provided")
+
+    try:
+        release_images = json.loads(release_images)
+    except json.JSONDecodeError as e:
+        raise ValueError("RELEASE_IMAGES enironment variable content is invalid JSON") from e
+
+    if len(release_images) == 0:
+        raise ValueError("RELEASE_IMAGES enironment variable content must contain at least one release")
+
+    latest_release_image = ReleaseImage(**release_images[-1])
+
+    if (
+        os_image := get_os_image(os_images=os_images, openshift_version=str(latest_release_image.openshift_version))
+    ) is None:
+        raise ValueError(
+            f"""no OS image found with openshift_version:
+            {latest_release_image.openshift_version} for the latest release image
+            """
+        )
+
+    os_images_dict = [os_image.to_dict()]
+
+    json.dump(os_images_dict, sys.stdout, separators=(",", ":"))
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("--src", required=True, type=str, help="OS images list file path")
```

Comment:
ambiguous name `src` 
maybe use `os_images_path` or `os_images_file`

Reply:
Not my change but I M ok with changing it